metaflow/runner/deployer.py

```diff
@@ -201,43 +195,38 @@ def run(self) -> Optional["metaflow.Run"]:
             return None
 
 
-class LazyDeploymentMethod:
-    def __init__(self, module_path, func_name):
-        self.module_path = module_path
-        self.func_name = func_name
-        self.func = None
-
-    def __call__(self, *args, **kwargs):
-        if self.func is None:
-            module = importlib.import_module(self.module_path)
-            self.func = getattr(module, self.func_name)
-        return self.func(*args, **kwargs)
-
-
-class DeploymentMethodsMeta(type):
-    from metaflow.plugins import FROM_DEPLOYMENT_PROVIDERS
-    from metaflow.metaflow_config import FROM_DEPLOYMENT_IMPL
-
+class DeployedFlowMeta(type):
     def __new__(mcs, name, bases, dct):
         cls = super().__new__(mcs, name, bases, dct)
 
-        def from_deployment(identifier, metadata=None, impl=None):
-            if impl is None:
-                impl = mcs.FROM_DEPLOYMENT_IMPL
+        from metaflow.plugins import DEPLOYER_IMPL_PROVIDERS
+
+        allowed_providers = dict(
+            {
+                provider.TYPE.replace("-", "_"): provider
+                for provider in DEPLOYER_IMPL_PROVIDERS
+            }
+        )
 
-            if impl not in mcs.FROM_DEPLOYMENT_PROVIDERS:
-                raise ValueError("This method is not available for: %s" % impl)
+        def _default_injected_method():
+            def f(
+                identifier: str,
+                metadata: Optional[str] = None,
+                impl: str = DEFAULT_FROM_DEPLOYMENT_IMPL.replace("-", "_"),
+            ):
+                if impl in allowed_providers:
+                    return allowed_providers[impl].DEPLOYED_FLOW_TYPE.from_deployment(
+                        identifier, metadata, impl
```

Comment:
I guess we don't need to pass `impl` to the underlying `from_deployment`

Reply:
Correct -- again, doing it for consistency so all methods have the same signature.

Reply:
I vote for removing it from the underlying implementation since it serves no purpose..
the previous implementation also didn't have it..

the injected function can have the `impl` and based on this value, the correct underlying implementation can be called without passing in the `impl`